builtin/providers/aws/config.go

```diff
@@ -195,23 +198,25 @@ func (c *Config) Client() (interface{}, error) {
 		dynamoSess := sess.Copy(&aws.Config{Endpoint: aws.String(c.DynamoDBEndpoint)})
 		kinesisSess := sess.Copy(&aws.Config{Endpoint: aws.String(c.KinesisEndpoint)})
 
-		// These two services need to be set up early so we can check on AccountID
-		client.iamconn = iam.New(awsIamSess)
-		client.stsconn = sts.New(sess)
+		if c.SkipIamValidation == false {
+			// These two services need to be set up early so we can check on AccountID
+			client.iamconn = iam.New(awsIamSess)
+			client.stsconn = sts.New(sess)
```

Comment:
This would cause Terraform to crash if you tried `apply`ing any `iam_*` resource with `SkipIamValidation` set to `true`. Is there any reason why not keep it outside of the condition?